plugins/outputs/newrelic/newrelic.go

```diff
@@ -0,0 +1,115 @@
+package newrelic
+
+// newrelic.go
+import (
+	"context"
+	"fmt"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/newrelic/newrelic-telemetry-sdk-go/cumulative"
+	"github.com/newrelic/newrelic-telemetry-sdk-go/telemetry"
+)
+
+// NewRelic nr structure
+type NewRelic struct {
+	harvestor    *telemetry.Harvester
+	dc           *cumulative.DeltaCalculator
+	InsightsKey  string `toml:"insights_key"`
+	MetricPrefix string `toml:"metric_prefix"`
+}
+
+// Description returns a one-sentence description on the Output
+func (nr *NewRelic) Description() string {
+	return "Send metrics to New Relic metrics endpoint"
+}
+
+// SampleConfig : return  default configuration of the Output
+func (nr *NewRelic) SampleConfig() string {
+	return `
+	## New Relic Insights API key (required)
+	insights_key = "insights api key"
+	#metric_prefix if defined, prefix's metrics name for easy identification (optional)
+	# metric_prefix = "Telegraf_"
+`
+}
+
+// Connect to the Output
+func (nr *NewRelic) Connect() error {
+	if nr.InsightsKey == "" {
+		return fmt.Errorf("InsightKey is a required for newrelic")
+	}
+	var err error
+	nr.harvestor, err = telemetry.NewHarvester(telemetry.ConfigAPIKey(nr.InsightsKey))
+	if err != nil {
+		return fmt.Errorf("unable to connect to newrelic %v", err)
+	}
+
+	nr.dc = cumulative.NewDeltaCalculator()
+	return nil
+}
+
+// Close any connections to the Output
+func (nr *NewRelic) Close() error {
+	nr.harvestor = nil
+	nr.dc = nil
+	return nil
+}
+
+// Write takes in group of points to be written to the Output
+func (nr *NewRelic) Write(metrics []telegraf.Metric) error {
+	for _, metric := range metrics {
+		// create tag map
+		tags := make(map[string]interface{})
+		for k, v := range metric.Tags() {
+			tags[k] = v
+		}
+		for k, v := range metric.Fields() {
```

Comment:
Update the Tags() and Fields() loops to use TagList() and FieldList(), these will perform much better.

Reply:
updated to use TagList() & FieldList()